// anujs/aaa/dist/pages/demo/native/button/index.js
import React from "../../../../ReactWX.js";

function P() {
  this.state = {
    defaultSize: 'default',
    primarySize: 'default',
    warnSize: 'default',
    disabled: false,
    plain: false,
    loading: false,
    ANU_ENV: "wx"
  };
}

P = React.toClass(P, React.Component, {
  setDisabled: function () {
    console.log('disabled');
    this.setState({
      disabled: !this.state.disabled
    });
  },
  setPlain: function () {
    this.setState({
      plain: !this.state.plain
    });
  },
  setLoading: function () {
    this.setState({
      loading: !this.state.loading
    });
  },
  getUserInfo: function (e) {
    console.log(e);
  },
  click: function (e) {
    console.log('click', e);
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "anu-block"
    }, this.state.ANU_ENV === 'quick' ? h("view", {
      class: "quick-container"
    }, h("text", {
      disabled: this.state.disabled,
      class: 'quick-button ' + (this.state.plain ? 'default-plain' : 'default'),
      onTap: this.click,
      "data-tap-uid": "e53_118",
      "data-beacon-uid": "default"
    }, "default"), h("text", {
      disabled: this.state.disabled,
      class: 'quick-button ' + (this.state.plain ? 'primary-plain' : 'primary'),
      onTap: this.click,
      "data-tap-uid": "e54_118",
      "data-beacon-uid": "default"
    }, "primary"), h("text", {
      disabled: this.state.disabled,
      class: 'quick-button ' + (this.state.plain ? 'warning-plain' : 'warning'),
      onTap: this.click,
      "data-tap-uid": "e55_118",
      "data-beacon-uid": "default"
    }, "warning"), h("text", {
      class: "quick-button default",
      onTap: this.setDisabled.bind(this),
      "data-tap-uid": "e58_41",
      "data-beacon-uid": "default"
    }, "点击设置以上按钮disabled属性"), h("text", {
      class: "quick-button default",
      onTap: this.setPlain.bind(this),
      "data-tap-uid": "e59_41",
      "data-beacon-uid": "default"
    }, "点击设置以上按钮plain属性")) : h("view", null, h("view", {
      style: "margin: 4px 0px"
    }, h("button", {
      class: "anu-block",
      type: "default",
      loading: this.state.loading,
      disabled: this.state.disabled,
      plain: this.state.plain
    }, "default")), h("view", {
      style: "margin: 4px 0px"
    }, h("button", {
      class: "anu-block",
      type: "primary",
      size: "mini",
      loading: this.state.loading,
      disabled: this.state.disabled,
      plain: this.state.plain
    }, "primary")), h("view", {
      style: "margin: 4px 0px"
    }, h("button", {
      class: "anu-block",
      type: "warn",
      disabled: this.state.disabled,
      plain: this.state.plain,
      loading: this.state.loading
    }, "warn"), h("button", {
      class: "anu-block",
      type: "warn",
      disabled: this.state.disabled,
      plain: this.state.plain
    }, "warn")), h("view", {
      style: "margin: 4px 0px"
    }, h("button", {
      onTap: this.setDisabled.bind(this),
      "data-tap-uid": "e100_15",
      "data-beacon-uid": "default"
    }, "点击设置以上按钮disabled属性")), h("view", {
      style: "margin: 4px 0px"
    }, h("button", {
      onTap: this.setPlain.bind(this),
      "data-tap-uid": "e103_15",
      "data-beacon-uid": "default"
    }, "点击设置以上按钮plain属性")), h("view", {
      style: "margin: 4px 0px"
    }, h("button", {
      onTap: this.setLoading.bind(this),
      "data-tap-uid": "e106_15",
      "data-beacon-uid": "default"
    }, "点击设置以上按钮loading属性")), h("button", {
      class: "bottom",
      type: "primary",
      "open-type": "getUserInfo",
      lang: "zh_CN",
      onGetUserInfo: this.getUserInfo.bind(this),
      "data-getuserinfo-uid": "e113_7",
      "data-beacon-uid": "default"
    }, "授权登录", ' '), ' '));;
  },
  classUid: "c3138"
}, {});
Page(React.registerPage(P, "pages/demo/native/button/index"));
export default P;
