// anujs/aaa/dist/components/Animal/index.js
import React from "../../ReactWX.js";

function Animal(props) {
  this.state = {
    name: props.name,
    age: props.age || 1
  };
}

Animal = React.toClass(Animal, React.Component, {
  changeAge: function () {
    this.setState({
      age: ~~(Math.random() * 10)
    });
  },
  componentDidMount: function () {
    console.log('Animal componentDidMount');
  },
  componentWillReceiveProps: function (props) {
    this.setState({
      name: props.name
    });
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      style: React.toStyle({
        border: '1px solid #333'
      }, this.props, 'style1372')
    }, "名字：", this.state.name, "年龄：", this.state.age, "岁", h("button", {
      catchTap: this.changeAge.bind(this),
      "data-tap-uid": "e36_24",
      "data-beacon-uid": "default"
    }, "换一个年龄"));;
  },
  classUid: "c906"
}, {});
Animal.defaultProps = {
  age: 1,
  name: 'animal'
};
Component(React.registerComponent(Animal, "Animal"));
export default Animal;
