// anujs/aaa/dist/pages/demo/native/slider/index.js
import React from "../../../../ReactWX.js";

function P() {
  this.bind();
}

P = React.toClass(P, React.Component, {
  bind: function () {
    for (let i = 1; i < 5; i++) {
      let index = i + 1;

      this['slider' + index + 'change'] = function (e) {
        console.log('slider' + 'index' + '发生 change 事件，携带值为', e.value);
      };
    }
  },
  switch1Change: function (e) {
    console.log('switch1 发生 change 事件，携带值为', e.value);
  },
  switch2Change: function (e) {
    console.log('switch2 发生 change 事件，携带值为', e.value);
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "anu-block"
    }, h("view", {
      class: "anu-item"
    }, h("text", {
      class: "anu-page-header"
    }, "设置step"), h("view", {
      class: "anu-block"
    }, h("slider", {
      onChange: this.slider2change,
      step: "5",
      "data-change-uid": "e39_32",
      "data-beacon-uid": "default"
    }))), h("view", {
      class: "anu-item"
    }, h("text", {
      class: "anu-page-header"
    }, "显示当前value"), h("view", {
      class: "anu-block"
    }, h("slider", {
      onChange: this.slider3change,
      "show-value": true,
      "data-change-uid": "e45_32",
      "data-beacon-uid": "default"
    }))), h("view", {
      class: "anu-item"
    }, h("text", {
      class: "anu-page-header"
    }, "设置最小/最大值"), h("view", {
      class: "anu-block"
    }, h("slider", {
      onChange: this.slider4change,
      min: "50",
      max: "200",
      "show-value": true,
      "data-change-uid": "e53_28",
      "data-beacon-uid": "default"
    }))), h("view", {
      class: "anu-item"
    }, h("view", null, h("switch", {
      checked: true,
      onChange: this.switch1Change,
      "data-change-uid": "e62_40",
      "data-beacon-uid": "default"
    }), h("switch", {
      onChange: this.switch2Change,
      "data-change-uid": "e63_32",
      "data-beacon-uid": "default"
    }))));;
  },
  classUid: "c2147"
}, {});
Page(React.registerPage(P, "pages/demo/native/slider/index"));
export default P;
