// anujs/aaa/dist/components/Calendar/index.js
import React from "../../ReactWX.js";
import { holidays } from "../../common/utils/holidayDate.js";
import EventEmitter from "../../common/utils/EventEmitter.js";
const TRAIN_MULTI_SELECT_SUB_TEXT = '备选';
const TRAIN_MULTI_SELECT_MAIN_TEXT = '主选';

function isEmptyObject(obj) {
  let arr = Object.keys(obj);
  return arr.length === 0;
}

function Calendar(props) {
  var now = new Date();
  var currentYear = now.getFullYear();
  var currentMonth = now.getMonth();
  var calendarDays = parseFloat(props.calendarDays) || 90;
  this.state = {
    eventType: props.eventType,
    storageType: props.storageType,
    sText: props.sText,
    eText: props.eText,
    dateTitle: ['日', '一', '二', '三', '四', '五', '六'],
    toViewId: props.toViewId,
    calendarArray: generateDates(currentYear, currentMonth, calendarDays, holidays, {}),
    isDoubleSelect: !!props.isDoubleSelect,
    isMultiSelect: !!props.isMultiSelect,
    heightStyle: {}
  };
  var that = this;
  React.api.getSystemInfo({
    success: function (sysInfo) {
      if (sysInfo.windowHeigh !== 320) {
        that.setState({
          heightStyle: {
            height: sysInfo.windowHeight + "px"
          }
        });
      }
    }
  });
  var formatToday = JSON.stringify(now).replace(/T.+|"/g, '');
  this.state.firstSelected = props.date || formatToday;

  if (props.isMultiSelect) {
    this.updateMultiSelectedDays(this.state, props);
  } else if (props.isDoubleSelect) {
    this.state.secondSelected = props.eDate;
  }

  this.updateCalendarData(this.state);
}

Calendar = React.toClass(Calendar, React.Component, {
  componentWillReceiveProps: function (nextProps) {
    if (!isEmptyObject(nextProps.updateInfoData)) {
      var now = new Date();
      var currentYear = now.getFullYear();
      var currentMonth = now.getMonth();
      var calendarDays = parseFloat(nextProps.calendarDays) || 90;
      this.setState({
        calendarArray: generateDates(currentYear, currentMonth, calendarDays, holidays, nextProps.updateInfoData)
      }, () => {
        this.updateCalendarData(this.state);
      });
    }
  },
  updateMultiSelectedDays: function (state, props) {
    var hash = {};
    var maxSelectDays = ~~props.maxSelectDays || 5;
    var multiSelectedDays = props.dates;

    if (!Array.isArray(multiSelectedDays)) {
      multiSelectedDays = [];
      state.days = multiSelectedDays;
    }

    multiSelectedDays.forEach(function (el) {
      hash[el] = 1;
    });
    state.maxSelectDays = maxSelectDays;
    state.multiSelectedDays = hash;
  },
  updateCalendarData: function (state, cb) {
    state.calendarArray.forEach(function (month) {
      month.daysArray.forEach(function (day) {
        calculateDayClassName(day, state);
      });
    });
    this.setState(state, cb);
  },
  dateSelected: function (item) {
    var isDisabled = item.isDisabled;

    if (isDisabled) {
      return;
    }

    if (this.state.isDoubleSelect) {
      this.dateSelectDouble(item);
    } else if (this.state.isMultiSelect) {
      this.dateSelectMulti(item);
    } else {
      this.dateSelectDefault(item);
    }
  },
  getNextDay: function (formateDate) {
    var throwIt = false;

    try {
      this.state.calendarArray.forEach(function (month) {
        month.daysArray.forEach(function (day) {
          if (day.formateDate === formateDate) {
            throwIt = true;
          } else if (throwIt && day.formateDate) {
            throw day.formateDate;
          }
        });
      });
    } catch (e) {
      return e;
    }
  },
  dateSelectDefault: function (item) {
    var state = this.state;
    var resdata = item.resdata;
    var {
      eventType,
      storageType
    } = this.state;
    state.firstSelected = state.firstSelected == item.formateDate ? null : item.formateDate;
    this.updateCalendarData(state, function () {
      this.dateSelectedFinish({
        eventType,
        storageType,
        resdata,
        showToast: false
      });
    });
  },
  dateSelectDouble: function (item) {
    var firstSelected = this.state.firstSelected;
    var secondSelected = this.state.secondSelected;
    var selectedDate = dateToNumber(item.formateDate);
    var curFirst = dateToNumber(firstSelected);
    var curSecond = dateToNumber(secondSelected);
    var {
      eventType,
      storageType
    } = this.state;

    if (this.state.confirmSecondDate) {
      if (selectedDate === curFirst || selectedDate === curSecond) {
        this.state.confirmSecondDate = false;
        this.updateCalendarData(this.state, () => {
          this.dateSelectedFinish({
            eventType,
            storageType
          });
        });
      } else if (selectedDate < curFirst) {
        this.state.firstSelected = item.formateDate;
        this.updateCalendarData(this.state);
      } else if (selectedDate > curFirst) {
        this.state.secondSelected = item.formateDate;
        this.state.confirmSecondDate = false;
        this.updateCalendarData(this.state, () => {
          this.dateSelectedFinish({
            eventType,
            storageType
          });
        });
      }
    } else {
      if (selectedDate < curFirst) {
        this.state.firstSelected = item.formateDate;

        if (!this.state.secondSelected) {
          this.state.secondSelected = this.state.eDate || this.getNextDay(item.formateDate);
        }
      } else if (selectedDate >= curFirst) {
        this.state.firstSelected = item.formateDate;
        this.state.secondSelected = this.getNextDay(item.formateDate);
        this.state.confirmSecondDate = true;
      }

      this.state.confirmSecondDate = true;
      this.updateCalendarData(this.state);
    }
  },
  dateSelectMulti: function (item) {
    var multiSelectedDays = this.state.multiSelectedDays || {};

    if (item.formateDate in multiSelectedDays) {
      delete multiSelectedDays[item.formateDate];
    } else {
      var maxSelectDays = this.state.maxSelectDays;

      if (Object.keys(multiSelectedDays).length >= maxSelectDays) {
        React.api.showToast({
          title: "最多可选择" + maxSelectDays + "个日期"
        });
        return;
      }

      this.state.multiSelectedDays[item.formateDate] = 1;
    }

    this.updateCalendarData(this.state);
  },
  dateSelectedFinish: function ({
    eventType,
    storageType = '',
    callbackData = '',
    resdata = '',
    showToast = true
  } = {}) {
    let bizType = this.state.bizType;
    let data = this.state;

    if (eventType) {
      let date;
      let storageDate;

      if (callbackData) {
        date = callbackData;
      } else if (data.isDoubleSelect) {
        date = {
          startDate: new Date(data.firstSelected),
          endDate: new Date(data.secondSelected)
        };
        storageDate = JSON.stringify(date);
      } else {
        storageDate = date = new Date(data.firstSelected);

        if (bizType === 'ticket' && resdata !== '') {
          date = JSON.parse(resdata);
          storageDate = resdata;
        }
      }

      EventEmitter.dispatch(eventType, date);

      if (storageType && storageType !== '') {
        try {
          React.api.setStorageSync(storageType, storageDate);
        } catch (e) {
          new Error(e);
        }
      }
    }

    if (showToast) {
      React.api.showToast({
        title: '日期选择成功',
        icon: 'success',
        duration: 500,
        success: function () {
          setTimeout(function () {
            React.api.navigateBack();
          }, 100);
        }
      });
    } else {
      setTimeout(() => {
        React.api.navigateBack();
      }, 100);
    }
  },
  multiSelectFinish: function () {
    var callbackData = Object.keys(this.state.multiSelectedDays);
    var eventType = this.state.eventType;
    this.dateSelectedFinish({
      eventType,
      callbackData
    });
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "calendar-content",
      style: React.toStyle(this.state.heightStyle, this.props, 'style19358')
    }, h("view", {
      className: "e-head"
    }, this.state.dateTitle.map(function (item, idx) {
      return h("text", {
        className: idx === 0 || idx === 6 ? 'w s' : 'w',
        key: item
      }, item);
    }, this)), h("scroll-view", {
      className: 'm-calendar  anu-col ' + (this.props.isMultiSelect ? 'multi' : ''),
      "scroll-y": "true",
      "scroll-into-view": this.state.toViewId
    }, this.state.calendarArray.map(function (month, i10185) {
      return h("view", {
        className: "e-month anu-col",
        id: 'cal_' + month.idMonth,
        type: month.title,
        key: month.title
      }, h("text", {
        className: "b-header anu-flex-center"
      }, month.title), h("view", {
        className: "b-row anu-row"
      }, month.daysArray.map(function (item, i10576) {
        return item.isBlank ? h("view", {
          class: "item",
          key: item.middleText
        }) : h("view", {
          key: item.middleText,
          onTap: this.dateSelected.bind(this, item),
          class: 'anu-flex-center anu-col-flex item item-a ' + item.className,
          "data-tap-uid": 'e289_44_' + i10185 + '-' + i10576,
          "data-beacon-uid": "default"
        }, h("text", {
          class: 'base ' + item.topClass
        }, item.topText), h("text", {
          class: 'dayclass ' + item.className
        }, item.middleText), item.bottomText && h("text", {
          class: item.isHighLight ? 'text highlight' : 'text'
        }, item.bottomText));
      }, this)));
    }, this)), this.props.isMultiSelect && h("view", null, h("view", {
      className: "e-btn-padding"
    }), h("view", {
      className: "e-btn",
      onTap: this.multiSelectFinish.bind(this),
      "data-tap-uid": "e309_44",
      "data-beacon-uid": "default"
    }, h("text", {
      className: "textColor"
    }, "确定"))));;
  },
  classUid: "c12677"
}, {});
Component(React.registerComponent(Calendar, "Calendar"));
export default Calendar;

function generateDates(currentYear, currentMonth, calendarDays, holidays, infoData) {
  var day = new Date(currentYear, currentMonth, 1).getDay();
  let monthLen = Math.floor(calendarDays / 30);
  var months = [];
  var formatToday = JSON.stringify(new Date()).replace(/T.+|"/g, '');
  var useEnabledCount = false,
      enabledCount = 0;

  for (var i = 0; i <= monthLen; i++) {
    var month = [];
    var curMonth = currentMonth + i;

    if (curMonth + i > 11) {
      currentYear++;
      curMonth = 0;
    }

    var formatMonth = curMonth + 1 < 10 ? '0' + (curMonth + 1) : curMonth + 1;
    month = {
      idMonth: currentYear + '-' + formatMonth,
      title: currentYear + '月' + formatMonth + '日',
      daysArray: []
    };

    if (day !== 0) {
      for (let j = 0; j < day; j++) {
        month.daysArray.push({
          isBlank: true,
          middleText: Math.random()
        });
      }
    }

    for (let j = 1; j <= 31; j++) {
      if (j > 27) {
        if (new Date(currentYear, curMonth, j).getMonth() !== curMonth) {
          continue;
        }
      }

      var formateDate = currentYear + '-' + formatMonth + '-' + (j < 10 ? '0' + j : j);
      var isDisabled = void 0;

      if (!useEnabledCount) {
        isDisabled = formateDate !== formatToday;

        if (isDisabled === false) {
          useEnabledCount = true;
          enabledCount = 1;
        }
      } else {
        enabledCount++;
        isDisabled = enabledCount > calendarDays;
      }

      var dateItem = {
        date: currentYear + '/' + curMonth + '/' + j,
        formateDate: formateDate,
        day: day,
        isBlank: false,
        topText: '',
        isWeekend: day === 0 || day === 6,
        middleText: formateDate == formatToday ? '今天' : j,
        bottomText: '',
        isDisabled: isDisabled,
        text: ''
      };

      if (holidays.holiday[formateDate]) {
        dateItem.topClass = 'holiday';
        dateItem.vacation = false;
        dateItem.topText = holidays.holiday[formateDate];
      }

      if (holidays.work[formateDate]) {
        dateItem.topClass = 'ban';
        dateItem.topText = '上班';
        dateItem.isWeekend = false;
      }

      if (holidays.vacation[formateDate]) {
        dateItem.topClass = 'xiu';
        dateItem.topText = holidays.holiday[formateDate] === 1 ? '放假' : holidays.holiday[formateDate];
        dateItem.vacation = true;
      }

      if (infoData[formateDate] && !isDisabled) {
        dateItem.text = infoData[formateDate].text || '';
        dateItem.isHighLight = infoData[formateDate].isHighLight || '';
      }

      day++;

      if (day == 7) {
        day = 0;
      }

      month.daysArray.push(dateItem);
    }

    months.push(month);
  }

  return months;
}

function calculateDayClassName(item, state) {
  if (item.formateDate === state.firstSelected && state.sText) {
    item.bottomText = state.sText;
  } else if (item.formateDate === state.secondSelected && state.eText) {
    item.bottomText = state.eText;
  } else if (item.text) {
    item.bottomText = item.text;
  } else {
    item.bottomText = '';
  }

  var classArray = [];

  if (state.isMultiSelect) {
    if (state.multiSelectedDays[item.formateDate]) {
      classArray.push('select');
      item.bottomText = TRAIN_MULTI_SELECT_SUB_TEXT;
    }

    if (item.formateDate === state.firstSelected) {
      classArray.push('select');
      item.bottomText = TRAIN_MULTI_SELECT_MAIN_TEXT;
    }
  } else if (item.formateDate === state.firstSelected) {
    classArray.push('select');
  } else if (item.formateDate === state.secondSelected) {
    classArray.push(state.confirmSecondDate ? 'select-second' : 'select');
  }

  if (item.isDisabled) {
    classArray.push('disabled');
  } else if (item.isWeekend || item.vacation) {
    classArray.push('weekend');
  }

  item.className = classArray.join(' ');
}

function dateToNumber(date) {
  if (!date) {
    return NaN;
  }

  return parseFloat(date.replace(/-/g, ''));
}
