// anujs/aaa/dist/pages/demo/apis/index.js
import React from "../../../ReactWX.js";

function alert(msg) {
  React.api.showModal({
    title: '',
    content: msg,
    showCancel: false,
    confirmText: '确定'
  });
}

function Express() {
  this.state = {
    text: 'page3',
    img: []
  };
}

Express = React.toClass(Express, React.Component, {
  componentDidMount: function () {},
  click: function () {
    console.log(111);
  },
  showModal: function () {
    React.api.showModal({
      title: '我是一个title',
      content: '内容是啥',
      cancelText: '取消',
      confirmText: '确定',
      success: function (result) {
        console.log('result', result);
      }
    });
  },
  showContextMenu: function () {
    React.api.showActionSheet({
      itemList: ['item1', 'item2', 'item3', 'item4'],
      itemColor: '#ff33ff',
      success: function (data) {
        alert("handling success, " + JSON.stringify(data));
      },
      cancel: function () {
        alert('handling cancel');
      },
      fail: function (data, code) {
        alert("handling fail, code = " + code);
      }
    });
  },
  vibrateLong: function () {
    console.log('vibrateLong');
    React.api.vibrateLong({
      success: () => {
        alert('震动起来了');
      }
    });
  },
  vibrateShort: function () {
    console.log('vibrateShort');
    React.api.vibrateShort({
      success: () => {
        alert('震动起来了');
      }
    });
  },
  upload: function () {
    console.log('upload');
    React.api.chooseImage({
      success: function (data) {
        console.log("handling success: " + data.uri);
        React.api.uploadFile({
          url: 'http://yapi.demo.qunar.com/mock/291/aaaaa',
          filePath: data.uri,
          name: 'file1',
          formData: {
            user: 'test'
          },
          success: function (data) {
            React.api.showModal({
              title: 'success'
            });
          },
          fail: function (data, code) {
            console.log("handling fail, code = " + code);
            React.api.showModal({
              title: 'fail',
              content: "code = " + code
            });
          }
        });
      }
    });
  },
  request: function () {
    React.api.request({
      url: 'http://yapi.demo.qunar.com/mock/13807/unauth/account/register',
      method: 'post',
      success: function (res) {
        alert("the status code of the response: " + JSON.stringify(res.data));
      },
      fail: function (err) {
        alert("handling fail, code = " + err);
      }
    });
  },
  download: function () {
    React.api.downloadFile({
      url: 'https://yapi.ymfe.org/devops/index.html#%E7%A6%81%E6%AD%A2%E6%B3%A8%E5%86%8C',
      success: function (data) {
        alert("handling success" + data);
      },
      fail: function (data, code) {
        alert("handling fail, code = " + code);
      }
    });
  },
  scan: function () {
    React.api.scanCode({
      success: function (data) {
        alert("handling success: " + data.result);
      },
      fail: function (data, code) {
        alert("handling fail, code = " + code);
      }
    });
  },
  onShareAppMessage: function () {
    var path = React.getCurrentPage().props.path;
    return {
      title: '妹子图片',
      path: path,
      imageUrl: "/assets/logo.jpg",
      success: res => {
        React.api.showToast({
          title: 'handling success' + res
        });
      },
      fail: res => {
        React.api.showToast({
          title: 'handling fail' + res
        });
      }
    };
  },
  getSavedFileInfo: function () {
    React.api.getSavedFileInfo({
      filePath: 'internal://Users/qitmac000476/Documents/weixin/nanachi/_site/apis/file.html',
      success: function (data) {
        alert("uri:" + data.uri + ", size: " + data.size + ", createTime: " + data.createTime);
      },
      fail: function (data, code) {
        alert("handling fail, code = " + code);
      }
    });
  },
  gotoSome: function (url) {
    console.log('url', url);

    if (url) {
      React.api.navigateTo({
        url
      });
    }
  },
  getLocation: function () {
    React.api.getLocation({
      success: function (data) {
        alert("handling success: longitude = " + data.longitude + ", latitude = " + data.latitude);
      }
    });
  },
  getNetworkType: function () {
    React.api.getNetworkType({
      success: function (data) {
        alert("handling success: " + data.networkType);
      }
    });
  },
  getSystemInfo: function () {
    React.api.getSystemInfo({
      success: function (data) {
        alert("handling success: " + JSON.stringify(data));
      }
    });
  },
  chooseImage: function () {
    React.api.chooseImage({
      count: 1,
      success: res => {
        this.setState({
          img: res.tempFilePaths
        });
      }
    });
  },
  setTitleBar: function () {
    React.api.setNavigationBarTitle({
      title: 'a new title',
      success: function () {
        alert('setTitleBar success');
      }
    });
  },
  makePhoneCall: function () {
    React.api.makePhoneCall({
      phoneNumber: '10086'
    });
  },
  showToast: function () {
    React.api.showToast({
      title: 'showToast' + React.api.hideToast
    });
  },
  createShortcut: function () {
    React.api.createShortcut();
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "api-page"
    }, h("view", {
      class: "anu-block"
    }, h("view", {
      onTap: this.showModal,
      class: "anu-item",
      "data-tap-uid": "e246_15",
      "data-beacon-uid": "default"
    }, h("text", null, "showModal")), h("view", {
      onTap: this.showToast,
      class: "anu-item",
      "data-tap-uid": "e249_15",
      "data-beacon-uid": "default"
    }, h("text", null, "showToast")), h("view", {
      onTap: this.showContextMenu,
      class: "anu-item",
      "data-tap-uid": "e252_15",
      "data-beacon-uid": "default"
    }, h("text", null, "显示上下文菜单")), h("view", {
      onTap: this.vibrateShort,
      class: "anu-item",
      "data-tap-uid": "e255_15",
      "data-beacon-uid": "default"
    }, h("text", null, "短震")), h("view", {
      onTap: this.vibrateLong,
      class: "anu-item",
      "data-tap-uid": "e258_15",
      "data-beacon-uid": "default"
    }, h("text", null, "长震")), h("view", {
      onTap: this.makePhoneCall,
      class: "anu-item",
      "data-tap-uid": "e261_15",
      "data-beacon-uid": "default"
    }, h("text", null, "打电话")), h("view", {
      onTap: this.upload,
      class: "anu-item",
      "data-tap-uid": "e264_15",
      "data-beacon-uid": "default"
    }, h("text", null, "文件上传")), h("view", {
      onTap: this.download,
      class: "anu-item",
      "data-tap-uid": "e267_15",
      "data-beacon-uid": "default"
    }, h("text", null, "文件下载")), h("view", {
      onTap: this.gotoSome.bind(this, '../../../pages/demo/apis/request/index'),
      class: "anu-item",
      "data-tap-uid": "e271_12",
      "data-beacon-uid": "default"
    }, h("text", null, "数据请求")), h("view", {
      onTap: this.scan,
      class: "anu-item",
      "data-tap-uid": "e276_15",
      "data-beacon-uid": "default"
    }, h("text", null, "扫一扫")), h("view", {
      onTap: this.gotoSome.bind(this, '../../../pages/demo/apis/storage/index'),
      class: "anu-item",
      "data-tap-uid": "e280_12",
      "data-beacon-uid": "default"
    }, h("text", null, "存储")), h("view", {
      onTap: this.getSavedFileInfo,
      class: "anu-item",
      "data-tap-uid": "e286_15",
      "data-beacon-uid": "default"
    }, h("text", null, "获取本地文件")), h("view", {
      onTap: this.gotoSome.bind(this, '../../../pages/demo/apis/clipboard/index'),
      class: "anu-item",
      "data-tap-uid": "e290_12",
      "data-beacon-uid": "default"
    }, h("text", null, "剪切板")), h("view", {
      onTap: this.gotoSome.bind(this, '../../../pages/demo/apis/canvas/index'),
      class: "anu-item",
      "data-tap-uid": "e296_12",
      "data-beacon-uid": "default"
    }, h("text", null, "画板")), h("view", {
      onTap: this.getLocation,
      class: "anu-item",
      "data-tap-uid": "e301_15",
      "data-beacon-uid": "default"
    }, h("text", null, "获取地理位置")), h("view", {
      onTap: this.getNetworkType,
      class: "anu-item",
      "data-tap-uid": "e304_15",
      "data-beacon-uid": "default"
    }, h("text", null, "获取网络类型")), h("view", {
      onTap: this.getSystemInfo,
      class: "anu-item",
      "data-tap-uid": "e307_15",
      "data-beacon-uid": "default"
    }, h("text", null, "获取系统信息(getSystemInfo)")), h("view", {
      onTap: this.chooseImage,
      class: "anu-item",
      "data-tap-uid": "e310_15",
      "data-beacon-uid": "default"
    }, h("text", null, "选择图片")), this.state.img.map(function (item, i7623) {
      return h("image", {
        src: item
      });
    }, this), h("view", {
      onTap: this.setTitleBar,
      class: "anu-item",
      "data-tap-uid": "e318_15",
      "data-beacon-uid": "default"
    }, h("text", null, "setTitleBar")), h("view", {
      onTap: this.createShortcut,
      class: "anu-item",
      "data-tap-uid": "e321_15",
      "data-beacon-uid": "default"
    }, h("text", null, "保存图标到桌面")), h("view", {
      onTap: this.gotoSome.bind(this, '../../../pages/demo/apis/route/index'),
      class: "anu-item",
      "data-tap-uid": "e325_12",
      "data-beacon-uid": "default"
    }, h("text", null, "路由"))));;
  },
  classUid: "c8404"
}, {});
Express.config = {
  navigationBarTextStyle: 'white',
  navigationBarBackgroundColor: '#0088a4',
  navigationBarTitleText: 'API',
  backgroundColor: '#eeeeee',
  backgroundTextStyle: 'light'
};
Page(React.registerPage(Express, "pages/demo/apis/index"));
export default Express;
