// anujs/aaa/dist/pages/demo/native/input/index.js
import React from "../../../../ReactWX.js";

function P() {
  this.state = {
    focus: false,
    inputValue: ''
  };
}

P = React.toClass(P, React.Component, {
  bindButtonTap: function () {
    this.setState({
      focus: true
    });
  },
  bindKeyInput: function (e) {
    this.setState({
      inputValue: e.target.value.trim()
    });
  },
  bindReplaceInput: function (e) {
    console.log(e);
    var value = e.target.value;
    var pos = e.cursor;

    if (pos != -1) {
      var left = e.target.value.slice(0, pos);
      pos = left;
    }

    return {
      value: value.replace(/11/g, '2'),
      cursor: pos
    };
  },
  bindHideKeyboard: function () {},
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "anu-block"
    }, h("view", {
      class: "anu-item anu-block"
    }, h("text", {
      class: "anu-black-title"
    }, "可以自动聚焦的input"), h("view", {
      class: "weui-cell weui-cell_input"
    }, h("input", {
      class: "weui-input",
      "auto-focus": true,
      placeholder: "将会获取焦点"
    }))), h("view", {
      class: "anu-item anu-block"
    }, h("text", {
      class: "anu-black-title"
    }, "控制最大输入长度的input"), h("view", {
      class: "weui-cell weui-cell_input"
    }, h("input", {
      class: "weui-input",
      maxlength: "10",
      placeholder: "最大输入长度为10"
    }))), h("view", {
      class: "anu-item anu-block"
    }, h("text", {
      class: "anu-black-title"
    }, "实时获取输入值：", this.state.inputValue), h("view", {
      class: "weui-cell weui-cell_input"
    }, h("input", {
      class: "weui-input",
      maxlength: "10",
      onInput: this.bindKeyInput,
      placeholder: "输入同步到view中",
      "data-input-uid": "e84_28",
      "data-beacon-uid": "default"
    }))), h("view", {
      class: "anu-item anu-block"
    }, h("text", {
      class: "anu-black-title"
    }, "控制输入的input"), h("view", {
      class: "weui-cell weui-cell_input"
    }, h("input", {
      class: "weui-input",
      onInput: this.bindReplaceInput,
      placeholder: "连续的两个1会变成2",
      "data-input-uid": "e94_28",
      "data-beacon-uid": "default"
    }))), h("view", {
      class: "anu-item anu-block"
    }, h("text", {
      class: "anu-black-title"
    }, "控制键盘的input"), h("view", {
      class: "weui-cell weui-cell_input"
    }, h("input", {
      class: "weui-input",
      onInput: this.bindHideKeyboard,
      placeholder: "输入123自动收起键盘",
      "data-input-uid": "e104_28",
      "data-beacon-uid": "default"
    }))), h("view", {
      class: "anu-item anu-block"
    }, h("text", {
      class: "anu-black-title"
    }, "数字输入的input"), h("view", {
      class: "weui-cell weui-cell_input"
    }, h("input", {
      class: "weui-input",
      type: "number",
      placeholder: "这是一个数字输入框"
    }))), h("view", {
      class: "anu-item anu-block"
    }, h("text", {
      class: "anu-black-title"
    }, "密码输入的input"), h("view", {
      class: "weui-cell weui-cell_input"
    }, h("input", {
      class: "weui-input",
      type: "password",
      placeholder: "这是一个密码输入框"
    }))), h("view", {
      class: "anu-item anu-block"
    }, h("text", {
      class: "anu-black-title"
    }, "带小数点的input"), h("view", {
      class: "weui-cell weui-cell_input"
    }, h("input", {
      class: "weui-input",
      type: "digit",
      placeholder: "带小数点的数字键盘"
    }))), h("view", {
      class: "anu-item anu-block"
    }, h("text", {
      class: "anu-black-title"
    }, "身份证输入的input"), h("view", {
      class: "weui-cell weui-cell_input"
    }, h("input", {
      class: "weui-input",
      type: "idcard",
      placeholder: "身份证输入键盘"
    }))), h("view", {
      class: "anu-item anu-block"
    }, h("text", {
      class: "anu-black-title"
    }, "控制占位符颜色的input"), h("view", {
      class: "weui-cell weui-cell_input"
    }, h("input", {
      class: "weui-input",
      "placeholder-style": "color:#F76260",
      placeholder: "占位符字体是红色的"
    }))));;
  },
  classUid: "c5782"
}, {});
P.defaultProps = {
  name: 'zhangsan'
};
Page(React.registerPage(P, "pages/demo/native/input/index"));
export default P;
