// anujs/aaa/dist/pages/demo/syntax/children/index.js
import React from "../../../../ReactWX.js";

function P() {
  this.state = {
    hidden: true
  };
}

P = React.toClass(P, React.Component, {
  toggleDialog: function () {
    this.setState({
      hidden: !this.state.hidden
    });
  },
  closeDialog: function () {
    this.setState({
      hidden: true
    });
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "anu-block"
    }, h("view", null, "演示组件标签包含其他内容"), h("view", {
      hidden: this.state.hidden
    }, h(React.useComponent, {
      is: "Dialog",
      "data-instance-uid": 'i28_20_' + 0
    }, h("view", null, this.state.title), h("view", null, "弹窗其他内容1"), h("view", null, "弹窗其他内容2"), h("view", null, h("button", {
      type: "default",
      size: "mini",
      onTap: this.closeDialog.bind(this),
      "data-tap-uid": "e36_32",
      "data-beacon-uid": "default"
    }, "关闭")))), h("view", null, h("button", {
      type: "primary",
      onTap: this.toggleDialog.bind(this),
      "data-tap-uid": "e44_43",
      "data-beacon-uid": "default"
    }, "点我")));;
  },
  classUid: "c1497"
}, {});
Page(React.registerPage(P, "pages/demo/syntax/children/index"));
export default P;
