// anujs/aaa/dist/pages/demo/native/scrollView/index.js
import React from "../../../../ReactWX.js";

function P() {
  this.order = ['red', 'yellow', 'blue', 'green', 'red'];
  this.state = {
    toView: 'red',
    scrollTop: 100
  };
}

P = React.toClass(P, React.Component, {
  upper: function (e) {
    console.log(e);
  },
  lower: function (e) {
    console.log(e);
  },
  scroll: function (e) {
    console.log(e);
  },
  tap: function () {
    for (var i = 0; i < this.order.length; ++i) {
      if (this.order[i] === this.state.toView) {
        this.setState({
          toView: this.order[i + 1]
        });
        break;
      }
    }
  },
  tapMove: function () {
    var _self = this;

    this.setState({
      scrollTop: _self.state.scrollTop + 10
    });
  },
  render: function () {
    var h = React.createElement;
    return h("view", null, h("view", {
      class: "section"
    }, h("view", {
      class: "section__title"
    }, "vertical scroll"), h("scroll-view", {
      "scroll-y": true,
      style: "height: 200px; overflow: hidden;",
      onScrolltoupper: this.upper,
      onScrolltolower: this.lower,
      onScroll: this.scroll,
      "scroll-into-div": this.state.todiv,
      "scroll-top": this.state.scrollTop,
      "data-scrolltoupper-uid": "e55_24",
      "data-beacon-uid": "default",
      "data-scrolltolower-uid": "e56_24",
      "data-scroll-uid": "e57_24"
    }, h("view", {
      type: "green",
      id: "green",
      class: "scroll-view-item bc_green"
    }), h("view", {
      type: "red",
      class: "scroll-view-item bc_red"
    }), h("view", {
      type: "yellow",
      class: "scroll-view-item bc_yellow"
    }), h("view", {
      type: "blue",
      class: "scroll-view-item bc_blue"
    })), h("view", {
      class: "btn-area"
    }, h("button", {
      size: "mini",
      onTap: this.tap,
      "data-tap-uid": "e68_44",
      "data-beacon-uid": "default"
    }, "click me to scroll into div", ' '), h("button", {
      size: "mini",
      onTap: this.tapMove,
      "data-tap-uid": "e71_44",
      "data-beacon-uid": "default"
    }, "click me to scroll"))), h("view", {
      class: "section section_gap"
    }, h("view", {
      class: "section__title"
    }, "horizontal scroll"), h("scroll-view", {
      class: "scroll-view_H",
      "scroll-x": true,
      style: "width: 100%"
    }, h("view", {
      type: "green",
      id: "green",
      class: "scroll-view-item_H bc_green"
    }), h("view", {
      type: "red",
      id: "red",
      class: "scroll-view-item_H bc_red"
    }), h("view", {
      type: "yellow",
      id: "yellow",
      class: "scroll-view-item_H bc_yellow"
    }), h("view", {
      type: "blue",
      id: "blue",
      class: "scroll-view-item_H bc_blue"
    }))));;
  },
  classUid: "c3135"
}, {});
Page(React.registerPage(P, "pages/demo/native/scrollView/index"));
export default P;
