// anujs/aaa/dist/pages/demo/apis/storage/index.js
import React from "../../../../ReactWX.js";

function alert(msg) {
  React.api.showModal({
    title: '',
    content: msg,
    showCancel: false,
    confirmText: '确定'
  });
}

function P() {
  this.state = {
    data: '1234455',
    copy: ''
  };
}

P = React.toClass(P, React.Component, {
  setStorage: function () {
    React.api.setStorage({
      key: 'v1',
      data: {
        key: 12
      },
      success: function (data) {
        alert('handling success');
      },
      fail: function (data, code) {
        alert("handling fail, code = " + code);
      }
    });
  },
  getStorage: function () {
    React.api.getStorage({
      key: 'key',
      success: function (res) {
        alert("handling success, " + res.data);
      },
      fail: function (data, code) {
        alert("handling fail, code = " + code);
      }
    });
  },
  removeStorage: function () {
    React.api.removeStorage({
      key: 'v1',
      success: function (data) {
        alert('handling success');
      },
      fail: function (data, code) {
        alert("handling fail, code = " + code);
      }
    });
  },
  setStorageSync: function () {
    let data = React.api.setStorageSync('key', 'value');
    console.log('data', data);
  },
  getStorageSync: function () {
    let data = React.api.getStorageSync('v1');
    console.log(111111111, React.api.getStorageSync + "");
    alert("handling success, " + data);
  },
  removeStorageSync: function () {
    React.api.removeStorageSync('key');
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "col"
    }, h("view", {
      onTap: this.setStorage,
      class: "item",
      "data-tap-uid": "e78_13",
      "data-beacon-uid": "default"
    }, h("text", null, "setStorage")), h("view", {
      onTap: this.getStorage,
      class: "item",
      "data-tap-uid": "e81_13",
      "data-beacon-uid": "default"
    }, h("text", null, "getStorage")), h("view", {
      onTap: this.removeStorage,
      class: "item",
      "data-tap-uid": "e84_13",
      "data-beacon-uid": "default"
    }, h("text", null, "removeStorage")), h("view", {
      onTap: this.setStorageSync,
      class: "item",
      "data-tap-uid": "e87_13",
      "data-beacon-uid": "default"
    }, h("text", null, "setStorageSync")), h("view", {
      onTap: this.getStorageSync,
      class: "item",
      "data-tap-uid": "e90_13",
      "data-beacon-uid": "default"
    }, h("text", null, "getStorageSync")), h("view", {
      onTap: this.removeStorageSync,
      class: "item",
      "data-tap-uid": "e93_13",
      "data-beacon-uid": "default"
    }, h("text", null, "removeStorageSync")));;
  },
  classUid: "c2410"
}, {});
Page(React.registerPage(P, "pages/demo/apis/storage/index"));
export default P;
