// anujs/aaa/dist/pages/demo/native/picker/index.js
import React from "../../../../ReactWX.js";

function P(props) {
  this.state = {
    show: false,
    index: 0,
    multiIndex: ['无脊柱动物', '扁性动物', '猪肉绦虫'],
    date: '2016-09-01',
    time: '12:01',
    region: ['广东省', '广州市', '海珠区'],
    multiArray: [{
      name: '无脊柱动物',
      sub: [{
        name: '扁性动物',
        sub: [{
          name: '猪肉绦虫'
        }, {
          name: '吸血虫'
        }]
      }, {
        name: '线形动物',
        sub: [{
          name: '蛔虫'
        }]
      }]
    }, {
      name: '脊柱动物',
      sub: [{
        name: '鱼',
        sub: [{
          name: '鲫鱼'
        }, {
          name: '带鱼'
        }]
      }]
    }],
    list1: ['iphone', '华为', '小米', '三星', '魅族', '锤子', 'oppo', 'vivo', 'iphone1', '华为1', '小米1', '三星1', '魅族1', '锤子1', 'oppo1', 'vivo1']
  };
}

P = React.toClass(P, React.Component, {
  showPicker: function () {
    console.log('showPicker');
    this.setState({
      show: true
    });
  },
  close: function () {
    console.log('close');
    this.setState({
      show: false
    });
  },
  change: function (e) {
    console.log('...', e);
    this.setState({
      index: e.value
    });
  },
  bindMultiPickerChange: function (e) {
    this.setState({
      multiIndex: e.value
    });
  },
  bindDateChange: function (e) {
    console.log('...bindDateChange', e);
    this.setState({
      date: e.value
    });
  },
  bindTimeChange: function (e) {
    this.setState({
      time: e.value
    });
  },
  bindRegionChange: function (e) {
    this.setState({
      region: e.value
    });
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "picker-demo"
    }, h("view", {
      class: "picker-title"
    }, h("text", {
      class: "title-text"
    }, "基本")), h("view", {
      class: "item-li"
    }, h("text", {
      class: "item-li-detail"
    }, "一列"), h("picker", {
      value: 0,
      onCancel: this.close.bind(this),
      onChange: this.change.bind(this),
      range: this.state.list1,
      "data-cancel-uid": "e136_24",
      "data-beacon-uid": "default",
      "data-change-uid": "e137_24"
    }, h("text", null, "当前选择：", this.state.list1[this.state.index]))), h("view", {
      class: "item-li"
    }, h("text", {
      class: "item-li-detail"
    }, "多列"), h("picker", {
      mode: "multiSelector",
      value: this.state.multiIndex,
      onCancel: this.close.bind(this),
      onChange: this.bindMultiPickerChange.bind(this),
      range: this.state.multiArray,
      "data-cancel-uid": "e148_24",
      "data-beacon-uid": "default",
      "data-change-uid": "e149_24"
    }, h("text", null, "当前选择：", this.state.multiIndex[0], ",", this.state.multiIndex[1], ",", ' ', this.state.multiIndex[2]))), h("view", {
      class: "item-li"
    }, h("text", {
      class: "item-li-detail"
    }, "日期选择器"), h("picker", {
      mode: "date",
      value: this.state.date,
      start: "2015-09-01",
      end: "2017-09-01",
      onCancel: this.close.bind(this),
      onChange: this.bindDateChange.bind(this),
      "data-cancel-uid": "e165_24",
      "data-beacon-uid": "default",
      "data-change-uid": "e166_24"
    }, h("text", null, "当前选择：", this.state.date))), h("view", {
      class: "item-li"
    }, h("text", {
      class: "item-li-detail"
    }, "时间选择器"), h("picker", {
      mode: "time",
      value: this.state.time,
      start: "09:01",
      end: "21:01",
      onCancel: this.close.bind(this),
      onChange: this.bindTimeChange.bind(this),
      "data-cancel-uid": "e178_24",
      "data-beacon-uid": "default",
      "data-change-uid": "e179_24"
    }, h("text", null, "当前选择：", this.state.time))), h("view", {
      class: "item-li"
    }, h("text", {
      class: "item-li-detail"
    }, "省市区选择器"), h("picker", {
      mode: "region",
      value: this.state.region,
      onCancel: this.close.bind(this),
      onChange: this.bindRegionChange.bind(this),
      "data-cancel-uid": "e189_24",
      "data-beacon-uid": "default",
      "data-change-uid": "e190_24"
    }, h("text", null, "当前选择：", this.state.region[0], ",", this.state.region[1], ",", this.state.region[2]))), h("view", {
      style: React.toStyle({
        height: '800px'
      }, this.props, 'style12109')
    }));;
  },
  classUid: "c6163"
}, {});
Page(React.registerPage(P, "pages/demo/native/picker/index"));
export default P;
