// anujs/aaa/dist/pages/about/index.js
import React from "../../ReactWX.js";

function About() {
  this.state = {
    name: 'qunar快应用',
    desc: '即点即用，让你省去下载安装的步骤，立即使用各类服务',
    serviceType: '工具类',
    subjectInfo: '北京趣拿软件科技有限公司',
    copyright: ''
  };
}

About = React.toClass(About, React.Component, {
  render: function () {
    var h = React.createElement;
    return h("view", {
      className: "demo-page"
    }, h("image", {
      id: "icon",
      src: "/assets/logo.png"
    }), h("view", {
      className: "name"
    }, this.state.name), h("view", {
      className: "name"
    }, "手机品牌：", this.props.query.brand), h("view", {
      className: "name"
    }, "快应用版本号：", this.props.query.version), h("view", {
      id: "tags"
    }, h("view", {
      className: "tag"
    }, "无安装"), h("view", {
      className: "gap"
    }, "|"), h("view", {
      className: "tag"
    }, "体积小"), h("view", {
      className: "gap"
    }, "|"), h("view", {
      className: "tag"
    }, "一步直达")), h("view", {
      id: "desc"
    }, this.state.desc), h("view", {
      className: "detail detail-first"
    }, h("view", {
      className: "detail-title"
    }, "服务类型"), h("view", {
      className: "detail-content"
    }, this.state.serviceType)), h("view", {
      className: "detail"
    }, h("view", {
      className: "detail-title"
    }, "主体信息"), h("view", {
      className: "detail-content"
    }, this.state.subjectInfo)), h("view", {
      className: "detail"
    }, h("view", {
      className: "detail-title"
    }, "页面参数"), h("view", {
      className: "detail-content"
    }, "param1:", this.props.query.param1, "param2:", this.props.query.param1)), h("view", {
      className: "btn"
    }, h("button", {
      type: "primary",
      size: "default",
      onTap: React.api.createShortcut,
      "data-tap-uid": "e50_58",
      "data-beacon-uid": "default"
    }, "创建快捷方式")), h("view", {
      id: "footer"
    }, this.state.copyright));;
  },
  classUid: "c2076"
}, {});
Page(React.registerPage(About, "pages/about/index"));
export default About;
