// anujs/aaa/dist/pages/demo/apis/canvas/index.js
import React from "../../../../ReactWX.js";

function Data() {
  this.state = {
    list: 'fillStyle,arc,strokeStyle,textBaseline,rect,globalAlpha'.split(',').map(function (name) {
      return {
        url: '../../../../pages/demo/apis/canvas' + "/" + name + "/index",
        name: name
      };
    })
  };
}

Data = React.toClass(Data, React.Component, {
  componentDidMount: function () {},
  gotoSome: function (url) {
    console.log('url', url);

    if (url) {
      React.api.navigateTo({
        url
      });
    }
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "anu-block"
    }, this.state.list.map(function (item, i775) {
      return h("view", {
        className: "anu-item",
        onTap: this.gotoSome.bind(this, item.url),
        "data-tap-uid": 'e36_38_' + i775,
        "data-beacon-uid": "default"
      }, h("text", null, item.name));
    }, this));;
  },
  classUid: "c1170"
}, {});
Page(React.registerPage(Data, "pages/demo/apis/canvas/index"));
export default Data;
