// anujs/aaa/dist/pages/demo/syntax/index/index.js
import React from "../../../../ReactWX.js";

function Express() {
  this.state = {
    title: '语法相关',
    pages: [{
      title: '继承',
      url: '/pages/demo/syntax/extend/index'
    }, {
      title: '无状态组件',
      url: '/pages/demo/syntax/stateless/index'
    }, {
      title: '条件语句',
      url: '/pages/demo/syntax/if/index'
    }, {
      title: '条件语句遇到组件',
      url: '/pages/demo/syntax/ifcom/index'
    }, {
      title: '一重循环',
      url: '/pages/demo/syntax/loop/index'
    }, {
      title: '二重循环',
      url: '/pages/demo/syntax/loop2/index'
    }, {
      title: '三重循环',
      url: '/pages/demo/syntax/loop3/index'
    }, {
      title: '循环里面交替使用两种不同组件（对话形式）',
      url: '/pages/demo/syntax/loop4/index'
    }, {
      title: '行内样式',
      url: '/pages/demo/syntax/inlineStyle/index'
    }, {
      title: '组件套嵌内容',
      url: '/pages/demo/syntax/children/index'
    }, {
      title: 'async/await',
      url: '/pages/demo/syntax/await/index'
    }, {
      title: '组件嵌套组件',
      url: '/pages/demo/syntax/multiple/index'
    }, {
      title: '一个页面多个相同组件',
      url: '/pages/demo/syntax/loop5/index'
    }, {
      title: 'webview',
      url: '/pages/demo/syntax/webview/index'
    }]
  };
}

Express = React.toClass(Express, React.Component, {
  componentWillMount: function () {
    console.log('syntax componentWillMount');
  },
  componentDidMount: function () {
    console.log('syntax componentDidMount');
  },
  gotoSome: function (url) {
    React.api.navigateTo({
      url
    });
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "anu-block"
    }, h("view", {
      class: "anu-page-header"
    }, this.state.title), this.state.pages.map(function (page, i2868) {
      return h("view", {
        class: "anu-item"
      }, h(React.useComponent, {
        "open-type": "navigate",
        "hover-class": "navigator-hover",
        url: page.url,
        is: "Navigator",
        "data-instance-uid": 'i95_28_' + i2868
      }, page.title));
    }, this));;
  },
  classUid: "c3308"
}, {});
Express.config = {
  'navigationBarTextStyle': 'white',
  'navigationBarBackgroundColor': '#0088a4',
  'navigationBarTitleText': 'Demo',
  'backgroundColor': '#ffffff',
  'backgroundTextStyle': 'light'
};
Page(React.registerPage(Express, "pages/demo/syntax/index/index"));
export default Express;
