// anujs/aaa/dist/pages/demo/syntax/loop2/index.js
import React from "../../../../ReactWX.js";

function P() {
  this.state = {
    title: '',
    key: '',
    trs: [[{
      title: 'aaa'
    }, {
      title: 'bbb'
    }, {
      title: 'ccc'
    }], [{
      title: 'ddd'
    }, {
      title: 'eee'
    }, {
      title: 'fff'
    }], [{
      title: 'ggg'
    }, {
      title: 'hhh'
    }, {
      title: 'iii'
    }]]
  };
}

P = React.toClass(P, React.Component, {
  getData: function (item, e) {
    this.setState({
      title: item.title,
      key: e.target.dataset.key
    });
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "anu-block"
    }, h("view", null, "测试循环中的事件，点击下方格子"), h("view", null, "title:", this.state.title, '  key ', this.state.key), h("view", {
      class: "anu-block"
    }, this.state.trs.map(function (item, i969) {
      return h("view", {
        class: "anu-line"
      }, item.map(function (el, i1122) {
        return h("view", {
          class: "loop2-cell",
          key: el.title + 'den',
          onTap: this.getData.bind(this, el),
          "data-tap-uid": 'e44_44_' + i969 + '-' + i1122,
          "data-beacon-uid": "default"
        }, h(React.useComponent, {
          content: el.title,
          is: "Fish",
          "data-instance-uid": 'i46_44_' + i969 + '-' + i1122
        }));
      }, this));
    }, this)));;
  },
  classUid: "c1925"
}, {});
Page(React.registerPage(P, "pages/demo/syntax/loop2/index"));
export default P;
