// anujs/aaa/dist/pages/demo/native/index/index.js
import React from "../../../../ReactWX.js";

function P() {
  this.state = {
    title: "内置组件",
    array: "button,checkbox,input,slider,picker,radio,textarea,label,audio,camera,image,video".split(",").map(function (name) {
      return {
        url: "/pages/demo/native" + "/" + name + "/index",
        name: name
      };
    })
  };
}

P = React.toClass(P, React.Component, {
  componentWillMount: function () {
    console.log("native componentWillMount");
  },
  componentDidMount: function () {
    console.log("native componentDidMount");
  },
  gotoSome: function (url) {
    if (url) {
      React.api.navigateTo({
        url
      });
    }
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "anu-block"
    }, h("view", {
      class: "anu-page-header"
    }, this.state.title), this.state.array.map(function (item, i1306) {
      return h("view", {
        class: "anu-item"
      }, h("navigator", {
        onTap: this.gotoSome.bind(this, item.url),
        "open-type": "navigate",
        "hover-class": "navigator-hover",
        url: item.url,
        "data-tap-uid": 'e50_32_' + i1306,
        "data-beacon-uid": "default"
      }, item.name));
    }, this));;
  },
  classUid: "c1935"
}, {});
P.config = {
  navigationBarTextStyle: "white",
  navigationBarBackgroundColor: "#0088a4",
  navigationBarTitleText: "内置组件",
  backgroundColor: "orange",
  backgroundTextStyle: "light"
};
Page(React.registerPage(P, "pages/demo/native/index/index"));
export default P;
