// anujs/aaa/dist/pages/demo/apis/clipboard/index.js
import React from "../../../../ReactWX.js";

function Data() {
  this.state = {
    data: '1234455',
    copy: ''
  };
}

Data = React.toClass(Data, React.Component, {
  copy: function () {
    console.log('copy');
    React.api.setClipboardData({
      data: this.state.data
    });
  },
  paste: function () {
    console.log('paste');
    React.api.getClipboardData({
      success: res => {
        console.log('data', res);
        this.setState({
          copy: res.data
        });
      }
    });
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "col"
    }, h("view", {
      class: "item"
    }, this.state.data), h("view", {
      class: "item",
      onTap: this.copy,
      "data-tap-uid": "e35_26",
      "data-beacon-uid": "default"
    }, "复制"), h("view", {
      class: "item",
      onTap: this.paste,
      "data-tap-uid": "e36_27",
      "data-beacon-uid": "default"
    }, "粘贴"), h("view", {
      class: "item"
    }, this.state.copy));;
  },
  classUid: "c884"
}, {});
Page(React.registerPage(Data, "pages/demo/apis/clipboard/index"));
export default Data;
