// anujs/aaa/dist/app.js
import React from "./ReactWX.js";

function Global(...args) {
  this.globalData = {
    ufo: 'ufo',
    buildType: "wx"
  };
}

Global = React.toClass(Global, React.Component, {
  onCollectLogs: function (dataset, type, node) {
    var page = React.getCurrentPage();

    if (!page) {
      return;
    }

    var path = page.props.path;
    var uuid = dataset.beaconUid;

    if (node) {
      var xpath = [];

      while (node.parentNode) {
        var index = node.parentNode.children.indexOf(node);
        xpath.unshift(index);
        node = node.parentNode;
      }

      uuid = xpath.join('/');
    }

    console.log('收集日志', path, type, uuid);
  },
  onGlobalShow: function () {
    console.log(React.getCurrentPage().props.path, 'onGlobalShow');
  },
  onGlobalLoad: function () {},
  onShowMenu: function (pageInstance, app) {},
  onLaunch: function () {
    console.log('App launched');
  },
  classUid: "c9135"
}, {});
Global.config = {
  window: {
    backgroundTextStyle: 'light',
    navigationBarTitleText: 'mpreact',
    navigationBarTextStyle: 'white'
  },
  tabBar: {
    color: '#929292',
    selectedColor: '#00bcd4',
    borderStyle: 'black',
    backgroundColor: '#ffffff',
    list: [{
      pagePath: 'pages/index/index',
      iconPath: '/assets/image/homepage_normal.png',
      selectedIconPath: '/assets/image/homepage_select.png',
      text: '首页',
      name: 'Home'
    }, {
      pagePath: 'pages/demo/question/index/index',
      iconPath: '/assets/image/question_normal.png',
      selectedIconPath: '/assets/image/question_select.png',
      text: '问答社区',
      name: 'questionAndAnswer'
    }, {
      pagePath: 'pages/demo/userCenter/index',
      iconPath: '/assets/image/uc_normal.png',
      selectedIconPath: '/assets/image/uc_select.png',
      text: '我的',
      name: 'My'
    }]
  }
};
export default App(new Global());
