// anujs/aaa/dist/pages/demo/native/swiper/index.js
import React from "../../../../ReactWX.js";

function P() {
  this.state = {
    background: ['demo-text-1', 'demo-text-2', 'demo-text-3'],
    indicatorDots: true,
    vertical: false,
    autoplay: false,
    circular: false,
    interval: 2000,
    duration: 500,
    previousMargin: 0,
    nextMargin: 0
  };
}

P = React.toClass(P, React.Component, {
  changeProperty: function (e) {
    var propertyName = e.currentTarget.dataset.propertyName;
    var newData = {};
    newData[propertyName] = e.value;
    this.setState(newData);
  },
  changeIndicatorDots: function () {
    this.setState({
      indicatorDots: !this.state.indicatorDots
    });
  },
  changeAutoplay: function () {
    this.setState({
      autoplay: !this.state.autoplay
    });
  },
  intervalChange: function (e) {
    this.setState({
      interval: e.value
    });
  },
  durationChange: function (e) {
    this.setState({
      duration: e.value
    });
  },
  render: function () {
    var h = React.createElement;
    return h("view", null, h("view", {
      class: "page-body"
    }, h("view", {
      class: "page-section page-section-spacing swiper"
    }, h("swiper", {
      "indicator-dots": this.state.indicatorDots,
      autoplay: this.state.autoplay,
      circular: this.state.circular,
      vertical: this.state.vertical,
      interval: this.state.interval,
      duration: this.state.duration,
      "previous-margin": this.state.previousMargin + 'px',
      "next-margin": this.state.nextMargin + 'px'
    }, this.state.background.map(function (item, i1860) {
      return h("swiper-item", null, h("view", {
        class: 'swiper-item ' + item
      }));
    }, this))), h("view", {
      class: "page-section"
    }, h("view", {
      class: "weui-cells weui-cells_after-title"
    }, h("view", {
      class: "weui-cell weui-cell_switch"
    }, h("view", {
      class: "weui-cell__bd"
    }, "指示点"), h("view", {
      class: "weui-cell__ft"
    }, h("switch", {
      checked: this.state.indicatorDots,
      onChange: this.changeProperty,
      "data-property-name": "indicatorDots",
      "data-change-uid": "e83_40",
      "data-beacon-uid": "default"
    }))), h("view", {
      class: "weui-cell weui-cell_switch"
    }, h("view", {
      class: "weui-cell__bd"
    }, "自动播放"), h("view", {
      class: "weui-cell__ft"
    }, h("switch", {
      checked: this.state.autoplay,
      onChange: this.changeProperty,
      "data-property-name": "autoplay",
      "data-change-uid": "e93_40",
      "data-beacon-uid": "default"
    }))), h("view", {
      class: "weui-cell weui-cell_switch"
    }, h("view", {
      class: "weui-cell__bd"
    }, "衔接滑动"), h("view", {
      class: "weui-cell__ft"
    }, h("switch", {
      checked: this.state.circular,
      onChange: this.changeProperty,
      "data-property-name": "circular",
      "data-change-uid": "e103_40",
      "data-beacon-uid": "default"
    }))), h("view", {
      class: "weui-cell weui-cell_switch"
    }, h("view", {
      class: "weui-cell__bd"
    }, "竖向"), h("view", {
      class: "weui-cell__ft"
    }, h("switch", {
      checked: this.state.vertical,
      onChange: this.changeProperty,
      "data-property-name": "vertical",
      "data-change-uid": "e113_40",
      "data-beacon-uid": "default"
    }))))), h("view", {
      class: "page-section page-section-spacing"
    }, h("view", {
      class: "page-section-title"
    }, h("text", null, "幻灯片切换时长(ms)"), h("text", {
      class: "info"
    }, this.state.duration)), h("slider", {
      value: this.state.duration,
      min: "500",
      max: "2000",
      onChange: this.changeProperty,
      "data-property-name": "duration",
      "data-change-uid": "e130_28",
      "data-beacon-uid": "default"
    }), h("view", {
      class: "page-section-title"
    }, h("text", null, "自动播放间隔时长(ms)"), h("text", {
      class: "info"
    }, this.state.interval)), h("slider", {
      value: this.state.interval,
      min: "2000",
      max: "10000",
      onChange: this.changeProperty,
      "data-property-name": "interval",
      "data-change-uid": "e141_28",
      "data-beacon-uid": "default"
    }), h("view", {
      class: "page-section-title"
    }, h("text", null, "前边距(px)"), h("text", {
      class: "info"
    }, this.state.changeProperty)), h("slider", {
      value: this.state.previousMargin,
      min: "0",
      max: "50",
      onChange: this.changeProperty,
      "data-property-name": "previousMargin",
      "data-change-uid": "e154_28",
      "data-beacon-uid": "default"
    }), h("view", {
      class: "page-section-title"
    }, h("text", null, "后边距(px)"), h("text", {
      class: "info"
    }, this.state.nextMargin)), h("slider", {
      value: this.state.nextMargin,
      min: "0",
      max: "50",
      onChange: this.changeProperty,
      "data-property-name": "nextMargin",
      "data-change-uid": "e165_28",
      "data-beacon-uid": "default"
    }))));;
  },
  classUid: "c7130"
}, {});
Page(React.registerPage(P, "pages/demo/native/swiper/index"));
export default P;
