// anujs/aaa/dist/pages/demo/native/textarea/index.js
import React from "../../../../ReactWX.js";

function P() {
  this.state = {
    focus: false
  };
}

P = React.toClass(P, React.Component, {
  bindTextAreaBlur: function (e) {
    console.log(e.value);
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "page-body column-layout"
    }, h("view", {
      class: "page-section column-layout"
    }, h("view", {
      class: "page-section-title"
    }, "输入区域高度自适应，不会出现滚动条"), h("view", {
      class: "textarea-wrp"
    }, h("textarea", {
      onBlur: this.bindTextAreaBlur,
      autoHeight: true,
      "data-blur-uid": "e24_34",
      "data-beacon-uid": "default"
    }))), h("view", {
      class: "page-section column-layout"
    }, h("view", {
      class: "page-section-title"
    }, "这是一个可以自动聚焦的textarea"), h("view", {
      class: "textarea-wrp"
    }, h("textarea", {
      autoFocus: true,
      style: "height: 200rpx"
    }))));;
  },
  classUid: "c1189"
}, {});
Page(React.registerPage(P, "pages/demo/native/textarea/index"));
export default P;
