// anujs/aaa/dist/pages/demo/apis/canvas/fillStyle/index.js
import React from "../../../../../ReactWX.js";

function P() {
  this.state = {};
}

P = React.toClass(P, React.Component, {
  onShow: function () {
    this.drawCanvas2();
  },
  drawCanvas2: function () {
    const ctx = React.api.createCanvasContext('fill', this);
    ctx.rect(10, 10, 100, 30);
    ctx.setFillStyle('yellow');
    ctx.fill();
    ctx.beginPath();
    ctx.rect(10, 40, 100, 30);
    ctx.setFillStyle('blue');
    ctx.fillRect(10, 70, 100, 30);
    ctx.rect(10, 100, 100, 30);
    ctx.setFillStyle('red');
    ctx.fill();
    ctx.draw();
  },
  componentDidMount: function () {
    console.log('fill Did Mount');
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "anu-block"
    }, h("canvas", {
      class: "content",
      id: "fill",
      style: React.toStyle({
        border: '1px solid #eeeeee',
        backgroundColor: '#ffffff'
      }, this.props, 'style1963')
    }));;
  },
  classUid: "c1108"
}, {});
Page(React.registerPage(P, "pages/demo/apis/canvas/fillStyle/index"));
export default P;
