// anujs/aaa/dist/pages/demo/syntax/stateless/index.js
import React from "../../../../ReactWX.js";

function P() {
  this.state = {
    a: 111,
    b: 222,
    styles: {
      a: {
        border: '1px solid red'
      }
    }
  };
}

P = React.toClass(P, React.Component, {
  changeA: function (e) {
    this.setState({
      a: ~~e.target.value
    });
  },
  changeB: function (e) {
    this.setState({
      b: ~~e.target.value
    });
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "anu-block"
    }, h("view", {
      class: "anu-page-header",
      style: React.toStyle(this.state.styles.a, this.props, 'style1351')
    }, "无狀态组件"), h("view", {
      class: "anu-block"
    }, h("view", null, "输入a(绑定onInput事件)"), h("input", {
      type: "number",
      style: "border: 1px solid gray;width:50%",
      value: this.state.a,
      onInput: this.changeA.bind(this),
      "data-input-uid": "e37_24",
      "data-beacon-uid": "default"
    })), h("view", {
      class: "anu-block"
    }, h("view", null, "输入b(绑定onChange事件)"), h("input", {
      type: "number",
      style: "border: 1px solid gray;width:50%",
      value: this.state.b,
      onInput: this.changeB.bind(this),
      "data-input-uid": "e46_24",
      "data-beacon-uid": "default"
    })), h("view", null, h(React.useComponent, {
      a: this.state.a,
      b: this.state.b,
      is: "Count",
      "data-instance-uid": 'i50_20_' + 0
    })));;
  },
  classUid: "c1665"
}, {});
Page(React.registerPage(P, "pages/demo/syntax/stateless/index"));
export default P;
