// anujs/aaa/dist/pages/demo/apis/canvas/textBaseline/index.js
import React from "../../../../../ReactWX.js";

function Data() {
  this.state = {};
}

Data = React.toClass(Data, React.Component, {
  onShow: function () {
    this.drawCanvas2();
  },
  drawCanvas2: function () {
    const ctx = React.api.createCanvasContext('myCanvas2', this);
    ctx.setStrokeStyle('red');
    ctx.moveTo(5, 375);
    ctx.lineTo(295, 375);
    ctx.stroke();
    ctx.setFontSize(20);
    ctx.setTextBaseline('top');
    ctx.fillText('top', 5, 375);
    ctx.setTextBaseline('middle');
    ctx.fillText('middle', 50, 375);
    ctx.setTextBaseline('bottom');
    ctx.fillText('bottom', 120, 375);
    ctx.setTextBaseline('normal');
    ctx.fillText('normal', 200, 375);
    ctx.draw();
  },
  componentDidMount: function () {},
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "anu-block"
    }, h("canvas", {
      class: "content",
      id: "myCanvas2"
    }));;
  },
  classUid: "c1036"
}, {});
Page(React.registerPage(Data, "pages/demo/apis/canvas/textBaseline/index"));
export default Data;
