// anujs/aaa/dist/common/utils/requester/requester.js
import React from "../../../ReactWX.js";
import Config from '../config/config.js';
import RequestManager from './requestManager.js';

function request(obj, initUserData) {
  const service = obj.service;
  const globalData = React.getApp().globalData;
  let param = obj.param || {};

  if (!service || !service.length) {
    return false;
  }

  let queryComponents = [];

  for (let key in param) {
    let value = param[key] || '';
    queryComponents.push(key + '=' + encodeURIComponent(value));
  }

  const {
    cookies
  } = initUserData;

  for (let key in cookies) {
    let value = cookies[key] || '';
    queryComponents.push(key + '=' + encodeURIComponent(value));
  }

  if (globalData.bd_origin) queryComponents.push("bd_origin=" + globalData.bd_origin);
  if (globalData.hd_origin) queryComponents.push("hd_origin=" + globalData.hd_origin);
  let url = (obj.host ? obj.host : Config.settings.requestDomain) + service + '?' + queryComponents.join('&');
  url = decodeURIComponent(url);
  let header = obj.header || {
    'Content-Type': 'application/json'
  };
  let req = {
    url: url,
    data: obj.data,
    header,
    method: obj.method && obj.method.toUpperCase() || 'GET',
    dataType: obj.dataType || 'json',
    success: obj.success,
    fail: obj.fail,
    complete: obj.complete
  };
  RequestManager.request(req);
  return true;
}

export default request;
