// anujs/aaa/dist/pages/demo/syntax/loop/index.js
import React from "../../../../ReactWX.js";

function P() {
  this.state = {
    array: [{
      name: '狗1'
    }, {
      name: '狗2'
    }, {
      name: '狗3'
    }, {
      name: '狗4'
    }]
  };
}

P = React.toClass(P, React.Component, {
  changeNumbers: function () {
    var array = this.state.array.concat();
    array.push({
      name: '狗' + (new Date() - 0)
    });
    this.setState({
      array: array
    });
  },
  render: function () {
    var h = React.createElement;
    return h("view", {
      class: "anu-block"
    }, h("view", {
      onTap: this.changeNumbers.bind(this),
      "data-tap-uid": "e38_21",
      "data-beacon-uid": "default"
    }, "演示单重循环，点这里改变数组的个数"), h("view", {
      class: "anu-block"
    }, this.state.array.map((el, i986) => h(React.useComponent, {
      name: el.name,
      key: el.name,
      is: "Dog",
      "data-instance-uid": 'i43_24_' + i986
    }))));;
  },
  classUid: "c1251"
}, {});
Page(React.registerPage(P, "pages/demo/syntax/loop/index"));
export default P;
